reference/test/components/bitvector.cpp

```diff
@@ -0,0 +1,103 @@
+// SPDX-FileCopyrightText: 2017 - 2025 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "core/components/bitvector.hpp"
+
+#include <memory>
+#include <random>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include "core/base/index_range.hpp"
+#include "core/components/bitvector.hpp"
+#include "core/test/utils.hpp"
+#include "reference/components/bitvector.hpp"
+
+
+template <typename IndexType>
+class Bitvector : public ::testing::Test {
+protected:
+    using index_type = IndexType;
+    using device_type = gko::bitvector<index_type>;
+    using storage_type = typename device_type::storage_type;
+    constexpr static auto block_size = device_type::block_size;
+    Bitvector()
+        : ref{gko::ReferenceExecutor::create()}, rng{67593}, sizes{0,    1,
+                                                                   2,    16,
+                                                                   31,   32,
+                                                                   33,   40,
+                                                                   63,   64,
+                                                                   65,   127,
+                                                                   128,  129,
+                                                                   1000, 1024,
+                                                                   2000}
+    {}
+
+    std::vector<index_type> create_random_values(index_type num_values,
+                                                 index_type size)
```

Comment:
this won't (always) return a vector with `num_values` entries. Maybe call it `max_num_values`.
(If you want to always return `num_values`, then you need to use iota+shuffle+take.)

Reply:
good suggestion, thanks!